File: bin/static/src/app/agent-portal/shared/model/core/risk.ts
import { VariableData } from './variable-data';
import { Coverage } from './coverage';
/**
* this object represents a risk of policy
*
* @author jtorres
*/
export class Risk {

    id: number;
    riskNumber: number;
    name: string;
    effectiveDate: Date;
    expirationDate: Date;
    group: number;
    variableDataRisk: VariableData[];
    coverages: Coverage[];
    childrenRisks: Risk[];
}

File: src/main/java/co/com/fsighttech/imp/issuing/bo/core/Risk.java
/**
 * 
 */
package co.com.fsighttech.imp.issuing.bo.core;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

/**
 * this object represents a risk
 * 
 * @author jtorres
 *
 */

@Getter
@Setter
public class Risk {

	private Long id;
	private Long riskNumber;
	private String name;
	private LocalDate effectiveDate;
	private LocalDate expirationDate;
	private Long group; // group a risk inside a risk category if needed.
	private List<VariableData> variableDataRisk;
	private List<Coverage> coverages = new ArrayList<>();
	private List<Risk> childrenRisks;

}
